CMakeLists.txt

```diff
@@ -50,12 +53,33 @@ endif()
 
 message(STATUS EXTRA_LIBRARIES: ${EXTRA_LIBRARIES})
 
+find_library(ZMQ_LIB zmq)
+
+############
+# Protobuf
+############
+
+if (NOT APPLE)
+  include(FindProtobuf)
+  set(Protobuf_USE_STATIC_LIBS ON)
+  find_package(Protobuf)
+  message(STATUS "Protobuf lib: ${Protobuf_LIBRARY}, inc: ${Protobuf_INCLUDE_DIR}, protoc: ${Protobuf_PROTOC_EXECUTABLE}")
+endif()
+
+############
+# LibUSB
+############
+
+find_library(usb_LIBRARY NAMES usb-1.0 libusb usb)
+set(LibUSB_LIBRARIES ${usb_LIBRARY})
+
 ############
 # Boost
 ############
 
 set(Boost_NO_BOOST_CMAKE 1)
 set(Boost_USE_MULTITHREADED ON)
+set(Boost_USE_STATIC_LIBS ON)
```

Comment:
On Ubuntu 22.04, I get this error with this field set to `ON`:

```
[100%] Linking CXX shared library libmonero-java.so
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libboost_serialization.a(basic_iarchive.o): warning: relocation against `_ZTVN5boost7archive6detail14basic_iarchiveE' in read-only section `.text'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libboost_chrono.a(chrono.o): relocation R_X86_64_PC32 against symbol `_ZN5boost6system6detail10cat_holderIvE24system_category_instanceE' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
```

But it works when set to `OFF`. Anything wrong with leaving it off?

Reply:
Hm, if I set it to `OFF`, it builds successfully, but the libraries built on Ubuntu 20.04 still can't be run on Ubuntu 22.04, showing an error from libboost_chrono.so.

Reply:
I think boost is compiled without -fPIC on ubuntu. I will look into  libboost_chrono.so error.

Reply:
Yeah, maybe it's not possible to build libs compatible with both versions then.

Reply:
@woodser , does the actual build only run on Ubuntu 20.04?

Shared libs have different versions on Ubuntu 20.04 and Ubuntu 22.04.
Ex:
```
ubuntu-20-04:~$ ls /lib/x86_64-linux-gnu/libssl.*
/lib/x86_64-linux-gnu/libssl.a  /lib/x86_64-linux-gnu/libssl.so  /lib/x86_64-linux-gnu/libssl.so.1.1

ubuntu-22-04:~$ ls /lib/x86_64-linux-gnu/libssl.*
/lib/x86_64-linux-gnu/libssl.a  /lib/x86_64-linux-gnu/libssl.so  /lib/x86_64-linux-gnu/libssl.so.3
```

Reply:
I don't know that the same build has ever worked on both platforms.

Was hoping the static libs with a shared library wrapper (for JNI) would be cross compatible.

Reply:
Hi @woodser .

I got haveno working with native wallet on ubuntu 20 and 22 but I have had to:
- use openssl and unbound static lib
- clone boost (1.71.0) and build it static with -fPIC (maybe it will become a module in monero-java)
- disable Trezor hardware wallet support on monero build (USE_DEVICE_TREZOR=OFF). Because I have had problems to run haveno with protobuf built static with -fPIC. Is it a problem?

OBS: I built monero/monero-cpp/monero-java on ubuntu 20.

Reply:
Its ok to go in this way?

Reply:
> Yes I think we may have to go this way if we want this static/shared library wrapper idea to work, since the goal is to make the libraries standalone and supported in JNI. Maybe this is the best way?

I think so. This is done with the last commit.
PS. I only tested this on linux. I dont have how testing on others platforms.